third_party/move/move-vm/runtime/src/native_functions.rs

```diff
@@ -209,19 +205,38 @@ impl<'b> NativeContext<'_, 'b> {
         self.interpreter.get_stack_frames(count)
     }
 
-    pub fn gas_balance(&self) -> InternalGas {
-        self.gas_balance
+    pub fn legacy_gas_budget(&self) -> InternalGas {
+        self.gas_meter.legacy_gas_budget_in_native_context()
     }
 
     pub fn use_heap_memory(&mut self, amount: u64) {
```

Comment:
this isn't a trait implementation, so can we also rename the method to indicate legacy?

Reply:
Renamed